src/utils/i18n.ts

```diff
@@ -1,30 +1,61 @@
 import i18n from 'i18next'
-import { initReactI18next } from 'react-i18next'
+import { initReactI18next, useTranslation } from 'react-i18next'
 import en from '../locales/en.json'
 import zh from '../locales/zh.json'
 import { storeCachedData, fetchCachedData } from './cache'
 import { AppCachedKeys } from '../constants/cache'
 
+export type LanuageType = 'en' | 'zh'
+
+export type I18nType = typeof i18n
+
+export type I18nInfoType = {
+  i18n: I18nType
+  currentLanguage: LanuageType
+  setLanguage: (lan: LanuageType) => void
+  toggleLanguage: () => void
+}
+
+const getDefaultLanguage = () => fetchCachedData<LanuageType>(AppCachedKeys.AppLanguage) ?? 'en'
+
 i18n.use(initReactI18next).init({
   resources: {
     en,
     zh,
   },
-  fallbackLng: 'en',
+  fallbackLng: getDefaultLanguage(),
   interpolation: {
     escapeValue: false,
   },
 })
 
-export const changeLanguage = (lan: 'en' | 'zh') => {
-  if (lan.indexOf('zh') !== -1) {
-    i18n.changeLanguage('zh')
-  } else {
-    i18n.changeLanguage('en')
+export function useI18n(): I18nInfoType {
+  const { t } = useTranslation()
+
+  const changeLanguage = (lan: LanuageType) => {
+    if (lan.indexOf('zh') !== -1) {
+      i18n.changeLanguage('zh')
+    } else {
+      i18n.changeLanguage('en')
+    }
+    storeCachedData(AppCachedKeys.AppLanguage, lan)
   }
-  storeCachedData(AppCachedKeys.AppLanguage, lan)
-}
 
-export const currentLanguage = (): 'en' | 'zh' | null => fetchCachedData<'en' | 'zh'>(AppCachedKeys.AppLanguage)
+  const setLanguage = (lan: LanuageType) => {
+    changeLanguage(lan)
+  }
 
-export default i18n
+  const currentLanguage = i18n.language as LanuageType
+  return {
+    i18n: { ...i18n, t },
+    currentLanguage,
+    setLanguage,
+    toggleLanguage: () => {
+      if (currentLanguage === 'en') {
+        setLanguage('zh')
+      } else {
+        setLanguage('en')
+      }
```

Comment:
What about `setLanguage(currentLanguage === 'en' ? 'zh' : 'en')`

Reply:
It declines the extensibility. Ternary should be reverted if a new language is supported(Adding a language is quite easy with AI now). Actually, a `switch-case` is preferable.

Reply:
But ternary can be used with a whitelist as following

```js
setLanguage(whitelist.find(currentLanguage) ?? fallbackLanguage)
```